src/java/picard/util/ClippingUtility.java

```diff
@@ -264,12 +265,10 @@ public static int findIndexOfClipSequence(final byte[] read, final byte[] adapte
             int mismatches = 0;
 
             for (int i = 0; i < length; ++i) {
-                if (!SequenceUtil.isNoCall(adapterSequence[i]) && !SequenceUtil.basesEqual(adapterSequence[i], read[start + i])) {
+                if (!SequenceUtil.basesEqual(adapterSequence[i], read[start + i])) {
```

Comment:
This is going to do the wrong thing in a couple of different cases:
1) Where one read is fine and the other has lots of N's at the end, this will cause IlluminaBasecallsToSam and ClipIlluminaAdapters to incorrectly trim both reads.
2) If a read has adapter sequence in it, but then goes to all Ns at the end, it will trim at the wrong position.  E.g. a read that has 60 good bases, then 60 bases of adapter, then 30 Ns, will get trimmed around base 138 instead of base 60.